skypy/halo/mass.py

```diff
@@ -1,38 +1,264 @@
-r'''Halo mass module.
-
-This module provides methods to sample the  masses of dark matter halos.
+r'''Halo mass sampler.
+This code samples halos from their mass function.
 
 Models
 ======
-
 .. autosummary::
    :nosignatures:
    :toctree: ../api/
 
    press_schechter
-
+   halo_mass_function
+   halo_mass_sampler
+   sheth_tormen_collapse_function
+   press_schechter_collapse_function
 '''
 
 import numpy as np
+from scipy import integrate
+from functools import partial
+from astropy import units as u
 
 from skypy.utils.random import schechter
 
+__all__ = [
+    'press_schechter',
+    'halo_mass_function',
+    'halo_mass_sampler',
+    'sheth_tormen_collapse_function',
+    'press_schechter_collapse_function',
+ ]
+
+
+def halo_mass_function(M, sigma, collapse_function, cosmology):
```

Comment:
```suggestion
def halo_mass_function(M, k, Pk, z, cosmology, collapse_function, params):
```
- `sigma` should be calculated inside this function, not passed as an argument containing data
- `collapse_function` should be a function, not an array of data. Then calculate `f_c=collapse_function(sigma, params)` inside this function. This will allow us to create e.g. `press_schechter_mass_function` using `partial`